near-contract-standards/src/fungible_token/implementation.rs

```diff
@@ -224,6 +224,8 @@ impl FungibleTokenResolver for FungibleToken {
         receiver_id: ValidAccountId,
         amount: U128,
     ) -> U128 {
+        //check: MUST only be called as .then() callback from this contract
+        assert_eq!(env::predecessor_account_id(), env::current_account_id());
```

Comment:
there is actually `assert_self();` shortcut in near_sdk.

Reply:
You can also use the `#[private]` macro above the method declaration, which adds this check.  Good catch @luciotato!

Reply:
Good catch. Let's switch to `#[private]` as @willemneal suggested